tests/Service/ProgrammesAggregationService/FindProgrammesChildrenByTypeTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Service\ProgrammesAggregationService;
+
+use BBC\ProgrammesPagesService\Domain\Entity\Programme;
+use InvalidArgumentException;
+
+/**
+ * @covers BBC\ProgrammesPagesService\Service\ProgrammesAggregationService::findProgrammesChildrenByType
+ * @group service
+ * @group programmesAggregation
+ */
+class FindProgrammesChildrenByTypeTest extends AbstractProgrammesAggregationTest
+{
+    /**
+     * @expectedException InvalidArgumentException
+     */
+    public function testCannotGetBrandTypesChildrenForAnyAncestry()
+    {
+        $this->service()->findProgrammesChildrenByType(
+            $this->createMock(Programme::class),
+            'Brand'
+        );
+    }
+
+    public function testFindChildrenByTypeCallsProperMethods()
+    {
+        $mockProgramme = $this->createMock(Programme::class);
+        $mockProgramme->method('getDbAncestryIds')->willReturn([11, 12]);
+
+        $this->mockRepository->expects($this->once())
+                             ->method('findProgrammesByAncestryAndType')
+                             ->willReturn([['type' => 'clip'], ['type' => 'clip']]);
+
+        $mappedResults = $this->service()->findProgrammesChildrenByType($mockProgramme, 'Clip');
+        $this->assertContainsOnlyInstancesOf('BBC\ProgrammesPagesService\Domain\Entity\Clip', $mappedResults);
```

Comment:
Use `$this->programmesFromDbData($dbData)` like in `FindChildrenSeriesByParentTest`